model/task_queue.go

```diff
@@ -155,34 +155,44 @@ func ValidateNewGraph(t *task.Task, tasksToBlock []string) error {
 
 func (self *TaskQueue) BlockTaskGroupTasks(spec TaskSpec, taskID string) error {
 	catcher := grip.NewBasicCatcher()
-	tasksToBlock := []string{}
-	for _, it := range self.Queue {
-		if spec.Group == it.Group && spec.BuildVariant == it.BuildVariant && spec.Version == it.Version {
-			tasksToBlock = append(tasksToBlock, it.Id)
-		}
-	}
+	tasksToBlock := []task.Task{}
 	t, err := task.FindOneId(taskID)
 	if err != nil {
 		return errors.Wrapf(err, "problem finding task %s", taskID)
 	}
 	if t == nil {
 		return errors.Errorf("found nil task %s", taskID)
 	}
+
+	p, err := FindProjectFromTask(t)
+	if err != nil {
+		return errors.Wrapf(err, "problem getting project for task %s", t.Id)
+	}
+	tg := p.FindTaskGroup(t.TaskGroup)
+	encounteredTask := false
+	for _, tgTask := range tg.Tasks {
+		if encounteredTask {
+			found, err := task.FindOne(task.ByVersionForNameAndVariant(t.Version, tgTask, t.BuildVariant))
+			if err != nil {
+				catcher.Add(errors.Wrapf(err, "problem finding task %s", tgTask))
+			}
+			if found == nil {
+				catcher.Add(errors.Errorf("found nil task %s", tgTask))
+				continue
+			}
+			tasksToBlock = append(tasksToBlock, *found)
+		}
+		if t.DisplayName == tgTask {
```

Comment:
If you find the task I think you want to have every task in the group in tasksToBlock right? Right now you're adding a potentially random subset of them

Reply:
I want all later tasks to be blocked. Task group tasks in a single-host task group are dispatched in the order defined in the configuration file. The idea here is to iterate over the tasks in the task group until we find the one that failed, then block all later tasks.